### PR TITLE
ACR better texture, APC9 SV2 and Visual bugs fixed

### PR DESCRIPTION
**Pull request by Braiam23**
## 🤔 What type of PR is this? (check all applicable)

- [X] 🍕 Addition
- [ ] ⌨️ Productivity
- [ ] 🐛 Bug Fix
- [X] 🔥 Optimization
- [X] ⚙️ Configuration
- [ ] 🌟 Quality Of Life
- [X] ✨ Enhancement
- [ ] 📝 Documentation

## 📝 Description

It is time to give texture to a mythical and old weapon in the mod, on the other hand it is the first test to see how the new weapons work, I doubt that it will break something, I have already tried it many times

## 🖼️ Screenshots/Recordings
Old texture:

New texture:
image

old APC9:

New APC9
image

Visual Bugs:

Fixed:

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [X] 🙅 no documentation needed

### CODE REVIEW

**Comment by Paneedah** [comment]
Could you please provide the Befores?

**Comment by Braiam23** [comment]
> Could you please provide the Befores?

I didn't understand how before?

**Comment by Paneedah** [comment]
Was there no textures before? What exactly did you change?

Or did you update the guns to the new Gun System?

**Comment by Braiam23** [comment]
> Was there no textures before? What exactly did you change?
> 
> Or did you update the guns to the new Gun System?

I updated the texture in the acr and in the apc9 the system was updated

**Comment by Paneedah** [comment]
Could you attach screenshots of what these guns looked like before your changes are applied?

**Comment by Braiam23** [comment]
> Could you attach screenshots of what these guns looked like before your changes are applied?

Yes ready

**Review by Desoroxxx — CHANGES_REQUESTED** [review]
Also missing changelog

**Comment by Desoroxxx** [comment]
By the way, next pull request it would be a really good idea to delete you entire fork and contact me on Discord to be added to the repository so you can use branches and stay up to date with master it would make everything must cleaner

File: src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java [hunk 1]
```diff
@@ -41,11 +48,27 @@ public Item createGun(CommonProxy commonProxy) {
 		.withFlashScale(() -> 0.6f)
 		.withFlashOffsetX(() -> 0.12f)
 		.withFlashOffsetY(() -> 0.15f)
-		.withShellCasingForwardOffset(0.05f)
-		.withShellCasingVerticalOffset(-0.03f)
-		.withInaccuracy(1f)
-//		.withShellCasingEjectEnabled(false)
+		.withInaccuracy(2f)
```
[the remark this answers is not in the document]
Comment (Braiam23) [reply src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java:51]: i was testing the Inaccuracy,

File: src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
+        Object Magazine;
+        return new Weapon.Builder()
 		
 		.withName("apc9")
-		//.withAmmoCapacity(30)
 		.withFireRate(0.85f)
 		.withRecoil(2f)
 		.withZoom(0.9f)
 		.withConfigGroup(GunConfigurationGroup.SMG)
+        .withShellType(Type.PISTOL)
```
[the remark this answers is not in the document]
Comment (Braiam23) [reply src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java:34]: ok i'll fix it right away

**Comment by Braiam23** [comment]
> By the way, next pull request it would be a really good idea to delete you entire fork and contact me on Discord to be added to the repository so you can use branches and stay up to date with master it would make everything must cleaner

sorry that's why I'm using the outdated mod files

**Review by Desoroxxx — APPROVED** [review]

**Review by Desoroxxx — CHANGES_REQUESTED** [review]
Missing changelog

**Comment by Braiam23** [comment]
> Missing changelog

ready

**Comment by Desoroxxx** [comment]
By the way about the APC, is it working fine are the attachment good how many customization options are there, and also when you are supposed to revamp the weapon for the new system it seems to be essentially the same

**Comment by Braiam23** [comment]
> By the way about the APC, is it working fine are the attachment good how many customization options are there, and also when you are supposed to revamp the weapon for the new system it seems to be essentially the same

what are you talking about?

**Comment by Desoroxxx** [comment]
> > By the way about the APC, is it working fine are the attachment good how many customization options are there, and also when you are supposed to revamp the weapon for the new system it seems to be essentially the same
> 
> what are you talking about? It does work well as a current renewed weapon, with respect to its customization it is related to the model and the APC9 model does not have any accessory that modifies the weapon and regarding whether to pass the weapon from system 1 to system 2 is different in a certain sense because the old code has more modification like the position of the weapon during proning, the recoil and movements of the old animations and that covers many lines

Okay but it would need to be revamped to bring it to the standard of quality, like it needs to have a lot of possible attachements etc...
Look at all the other weapons that use the new system versus the old to see the quality difference

**Comment by Braiam23** [comment]
> > > By the way about the APC, is it working fine are the attachment good how many customization options are there, and also when you are supposed to revamp the weapon for the new system it seems to be essentially the same
> > 
> > 
> > what are you talking about? It does work well as a current renewed weapon, with respect to its customization it is related to the model and the APC9 model does not have any accessory that modifies the weapon and regarding whether to pass the weapon from system 1 to system 2 is different in a certain sense because the old code has more modification like the position of the weapon during proning, the recoil and movements of the old animations and that covers many lines
> 
> Okay but it would need to be revamped to bring it to the standard of quality, like it needs to have a lot of possible attachements etc... Look at all the other weapons that use the new system versus the old to see the quality difference

add more content to the weapon?

**Comment by Desoroxxx** [comment]
> > > > By the way about the APC, is it working fine are the attachment good how many customization options are there, and also when you are supposed to revamp the weapon for the new system it seems to be essentially the same
> > > 
> > > 
> > > what are you talking about? It does work well as a current renewed weapon, with respect to its customization it is related to the model and the APC9 model does not have any accessory that modifies the weapon and regarding whether to pass the weapon from system 1 to system 2 is different in a certain sense because the old code has more modification like the position of the weapon during proning, the recoil and movements of the old animations and that covers many lines
> > 
> > 
> > Okay but it would need to be revamped to bring it to the standard of quality, like it needs to have a lot of possible attachements etc... Look at all the other weapons that use the new system versus the old to see the quality difference
> 
> add more content to the weapon? Or do you mean the texture?

Both

**Comment by Braiam23** [comment]
> Both

Honestly, I would like to add more weapon accessories, but there is no time, as far as the texturing is concerned, it is fine because it is similar to the current style of the mod.

**Comment by Desoroxxx** [comment]
The APC9 seems to be way closer to the camera is that intended

**Pull request merged**